### PR TITLE
add module gatk4/shiftfasta

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

Closes https://github.com/nf-core/modules/issues/2367

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/gatk4/shiftfasta/main.nf [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    conda (params.enable_conda ? "bioconda::gatk4=4.3.0.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gatk4:4.3.0.0--py36hdfd78af_0':
+        'quay.io/biocontainers/gatk4:4.3.0.0--py36hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(fasta)
+    path (fasta_fai)
+    path (seq_dict)
```
Comment (nvnieuwk) [inline modules/nf-core/gatk4/shiftfasta/main.nf:13]:
```suggestion
    path (dict)
```

File: modules/nf-core/gatk4/shiftfasta/main.nf [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    tuple val(meta), path("*.shifted.intervals") , emit: shift_intervals   , optional: true
+    path "versions.yml"                          , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def dict = seq_dict ? "--sequence-dictionary ${seq_dict}" : ""
```
Comment (nvnieuwk) [inline modules/nf-core/gatk4/shiftfasta/main.nf:30]:
```suggestion
    def seq_dict = dict ? "--sequence-dictionary ${dict}" : ""
```

File: modules/nf-core/gatk4/shiftfasta/main.nf [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    } else {
+        avail_mem = task.memory.giga
+    }
+    """
+    gatk --java-options "-Xmx${avail_mem}g" ShiftFasta \\
+        --reference $fasta \\
+        --output ${prefix}_shift.fasta \\
+        --shift-back-output ${prefix}_shift.back_chain \\
+        $args \\
+        $dict \\
```
Comment (nvnieuwk) [inline modules/nf-core/gatk4/shiftfasta/main.nf:43]:
```suggestion
        $seq_dict \\
```

File: modules/nf-core/gatk4/shiftfasta/main.nf [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def dict = seq_dict ? "--sequence-dictionary ${seq_dict}" : ""
+    def avail_mem = 3
+    if (!task.memory) {
+        log.info '[GATK ApplyBQSR] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
```
Comment (nvnieuwk) [inline modules/nf-core/gatk4/shiftfasta/main.nf:33]:
```suggestion
        log.info '[GATK ShiftFasta] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
```

File: modules/nf-core/gatk4/shiftfasta/meta.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        e.g. [ id:'test' ]
+  - fasta:
+      type: file
+      description: fasta file
+      pattern: "*.{fa,fasta}"
+  - fasta_fai:
+      type: file
+      description: index for fasta file
+      pattern: "*.{fai}"
+  - seq_dict:
```
Comment (nvnieuwk) [inline modules/nf-core/gatk4/shiftfasta/meta.yml:32]:
```suggestion
  - dict:
```

File: modules/nf-core/gatk4/shiftfasta/meta.yml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+      pattern: "versions.yml"
+  - shift_fa:
+      type: file
+      description: Shifted fasta file
+      pattern: "*.{fa,fasta}"
+  - shift_back_chain:
+      type: file
+      description: The shiftback chain file to use when lifting over
+      pattern: "*.{back_chain}"
+  - intervals:
```
Comment (nvnieuwk) [inline modules/nf-core/gatk4/shiftfasta/meta.yml:55]: Can you also add the `dict` output here?

**Review by nvnieuwk — CHANGES_REQUESTED** [review]
Some minor cosmetic changes and a note on the `meta.yml`

**Review by nvnieuwk — APPROVED** [review]

**Comment by ramprasadn** [comment]
Thanks for taking a look at this PR @nvnieuwk. I have incorporated your suggestions now 👍🏻

**Pull request merged**